crates/ty_python_semantic/src/semantic_index/place.rs

```diff
@@ -1,1082 +1,563 @@
-            |id| self.table.places[*id].expr == place_expr.expr,
-            |id| PlaceTable::hash_place_expr(&self.table.places[*id].expr),
-        );
-
-        match entry {
-            Entry::Occupied(entry) => (*entry.get(), false),
-            Entry::Vacant(entry) => {
-                place_expr.expr.sub_segments.shrink_to_fit();
-                let id = self.table.places.push(place_expr);
-                entry.insert(id);
-                let new_id = self.associated_place_ids.push(SmallVec::new_const());
-                debug_assert_eq!(new_id, id);
-                for root in self.table.places[id].expr.root_exprs() {
-                    if let Some(root_id) = self.table.place_id_by_expr(root) {
-                        self.associated_place_ids[root_id].push(id);
-                    }
-                }
-                (id, true)
-            }
-        }
-    }
-
-    pub(super) fn mark_place_bound(&mut self, id: ScopedPlaceId) {
-        self.table.places[id].insert_flags(PlaceFlags::IS_BOUND);
-    }
-
-    pub(super) fn mark_place_declared(&mut self, id: ScopedPlaceId) {
-        self.table.places[id].insert_flags(PlaceFlags::IS_DECLARED);
-    }
-
-    pub(super) fn mark_place_used(&mut self, id: ScopedPlaceId) {
-        self.table.places[id].insert_flags(PlaceFlags::IS_USED);
-    }
-
-    pub(super) fn mark_place_global(&mut self, id: ScopedPlaceId) {
-        self.table.places[id].insert_flags(PlaceFlags::MARKED_GLOBAL);
-    }
-
-    pub(super) fn mark_place_nonlocal(&mut self, id: ScopedPlaceId) {
-        self.table.places[id].insert_flags(PlaceFlags::MARKED_NONLOCAL);
-    }
-
-    pub(super) fn places(&self) -> impl Iterator<Item = &PlaceExprWithFlags> {
-        self.table.places()
-    }
-
-    pub(super) fn place_id_by_expr(&self, place_expr: &PlaceExpr) -> Option<ScopedPlaceId> {
-        self.table.place_id_by_expr(place_expr)
+impl FilePlaceId {
+    pub fn scope(self) -> FileScopeId {
+        self.scope
     }
 
-    pub(super) fn place_expr(&self, place_id: impl Into<ScopedPlaceId>) -> &PlaceExprWithFlags {
-        self.table.place_expr(place_id)
+    pub(crate) fn scoped_place_id(self) -> ScopedPlaceId {
+        self.scoped_place_id
     }
+}
 
-    /// Returns the place IDs associated with the place (e.g. `x.y`, `x.y.z`, `x.y.z[0]` for `x`).
-    pub(super) fn associated_place_ids(
-        &self,
-        place: ScopedPlaceId,
-    ) -> impl Iterator<Item = ScopedPlaceId> {
-        self.associated_place_ids[place].iter().copied()
+impl From<FilePlaceId> for ScopedPlaceId {
+    fn from(val: FilePlaceId) -> Self {
+        val.scoped_place_id()
     }
+}
 
-    pub(super) fn finish(mut self) -> PlaceTable {
-        self.table.shrink_to_fit();
-        self.table
-    }
+pub(crate) struct ParentPlaceIter<'a> {
+    state: Option<ParentPlaceIterState<'a>>,
 }
 
-/// Reference to a node that introduces a new scope.
-#[derive(Copy, Clone, Debug)]
-pub(crate) enum NodeWithScopeRef<'a> {
```

Comment:
I moved everything *Scope* to `scope.rs`. That code doesn't need to belong into `place`.